src/main/scala/is/hail/stats/RegressionUtils.scala

```diff
@@ -265,9 +266,25 @@ object RegressionUtils {
 
     lrb.stats(y, nSamples, minAC)
   }
+
+  // constructs SparseVector of hard call genotypes (with missing values mean-imputed) in parallel with other summary statistics
+  // if all genotypes are missing then all elements are NaN
+  def toLinMixedHardCallStats(gs: Iterable[Genotype], mask: Array[Boolean], nSamples: Int): SparseGtVectorAndStats = {
+    val sb = new SparseGtBuilder()
+    val gts = gs.hardCallGenotypeIterator
+
+    var i = 0
+    while (i < mask.length) {
```

Comment:
I'm inclined to recalculate `nSamples` here during the loop because it seems tremendously easy to misunderstand the properties that must be true of `nSamples` and `mask` (namely that `mask.map(if (_) 1 else 0).sum == nSamples`). At the least, that invariant should be documented in the comments.